src/facebook/facebook-notifications/facebooknotificationsyncadaptor.cpp

```diff
@@ -87,11 +87,17 @@ void FacebookNotificationSyncAdaptor::requestNotifications(int accountId, const
     queryItems.append(QPair<QString, QString>(QString(QLatin1String("locale")), QLocale::system().name()));
     QUrl url(graphAPI(QLatin1String("/me/notifications")));
     if (pagingToken.isEmpty()) {
-        int sinceSpan = m_accountSyncProfile
-                      ? m_accountSyncProfile->key(Buteo::KEY_SYNC_SINCE_DAYS_PAST, QStringLiteral("7")).toInt()
-                      : 7;
+        QDateTime since = lastSyncTimestamp(QLatin1String("facebook"),
```

Comment:
We really need to rewrite these plugins, and throw away all of the SocialNetworkSyncAdaptor baseclass code.  This since timestamp will be set even if the sync failed (because it's set in decrementSemaphore()) - so in the case where sync is aborted due to network loss or whatever, we probably need special handling to set the sync timestamp appropriately.  See https://github.com/nemomobile/buteo-sync-plugins-social/pull/42/files#diff-87eaa163286f08d3761545a3286793c8R800 for an example.